### PR TITLE
feature(generator): parallelize code generation

### PR DESCRIPTION
**Pull request by scotthart**
Recent changes to ServiceCodeGenerator, to guarantee immutability after construction, allows for safe generation of all code files concurrently.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5466)

### CODE REVIEW

File: generator/generator.cc [hunk 1]
```diff
@@ -54,15 +55,26 @@ bool Generator::Generate(google::protobuf::FileDescriptor const* file,
...
+      tasks.push_back(
+          std::async(std::launch::async, [&c]() { return c->Generate(); }));
     }
   }
+
+  std::string error_message;
+  for (auto& t : tasks) {
+    auto result = t.get();
+    if (!result.ok()) {
+      error_message += result.message();
```
Comment (devbww) [inline generator/generator.cc:70]: Do you need to delineate multiple error messages somehow?

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5466)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @devbww)

___
*[generator/generator.cc, line 70 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5466#-MLnIyGlBPZ-vJnSLWVB-r1-70:-MLnWA6f56n6JXOMCV5z:bcjzjex) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/855d74b96fad0ed4ca5fb98f3498af979effaf5b/generator/generator.cc#L70)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

Do you need to delineate multiple error messages somehow?
</blockquote></details>

The error messages themselves should contain all the relevant/identifying information.

**Review by devbww — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5466)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @scotthart)

___
*[generator/generator.cc, line 70 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5466#-MLnIyGlBPZ-vJnSLWVB-r1-70:-MLn__sq6CnoIvDpS10X:bl5lp6w) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/855d74b96fad0ed4ca5fb98f3498af979effaf5b/generator/generator.cc#L70)):*
<details><summary><i>Previously, scotthart (Scott Hart) wrote…</i></summary><blockquote>

The error messages themselves should contain all the relevant/identifying information.
</blockquote></details>

I meant that if one task fails with "cannot foo" and another fails with "cannot bar" then this will fail with "cannot foocannot bar", which looks confusing.

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5466)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @devbww)

___
*[generator/generator.cc, line 70 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5466#-MLnIyGlBPZ-vJnSLWVB-r1-70:-MLnf-1O8KQzHFNsUE0S:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/855d74b96fad0ed4ca5fb98f3498af979effaf5b/generator/generator.cc#L70)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

I meant that if one task fails with "cannot foo" and another fails with "cannot bar" then this will fail with "cannot foocannot bar", which looks confusing.
</blockquote></details>

Done.

**Review by devbww — APPROVED** [review]
> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5466)* status: :shipit: complete! all files reviewed, all discussions resolved

**Pull request merged**